tools/deployment/linux_pre_uninstall.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+#  Copyright (c) 2014-present, Facebook, Inc.
+#  All rights reserved.
+#
+#  This source code is licensed under the BSD-style license found in the
+#  LICENSE file in the root directory of this source tree. An additional grant
+#  of patent rights can be found in the PATENTS file in the same directory.
+
+SCRIPT_DIR="/usr/share/osquery/scripts"
+
+INITD_SRC="$SCRIPT_DIR/osqueryd.initd"
+INITD_DIR="/etc/rc.d/init.d"
+INITD_DIR_DEB="/etc/init.d"
+
+SYSTEMD_SERVICE_SRC="$SCRIPT_DIR/osqueryd.service"
+SYSTEMD_SERVICE_DIR="/usr/lib/systemd/system"
+
+SYSTEMD_SYSCONFIG_SRC="$SCRIPT_DIR/osqueryd.sysconfig"
+SYSTEMD_SYSCONFIG_DIR="/etc/sysconfig"
+SYSTEMD_SYSCONFIG_DIR_DEB="/etc/default"
+
+# Check if OS is Debian based
+if ls /etc | grep -Fq debian_version; then
+  INITD_DIR=$INITD_DIR_DEB
+  SYSTEMD_SYSCONFIG_DIR=$SYSTEMD_SYSCONFIG_DIR_DEB
+fi
+
+# Check if system is using upstart
+if ps -p 1 | grep -Fq init && init --version | grep -Fq upstart; then
+  if [ -f $INITD_SRC ] && [ -f $INITD_DIR/osqueryd ]; then
+    if [ "`diff -q $INITD_SRC $INITD_DIR/osqueryd`" ]; then
```

Comment:
Two things:
- normally init scripts and systemd units aren't meant to be edited by the user (especially for systemd, overriding them is trivial anyway)
- if you still want to make the init.d user editable, you can mark it as a conffile and both dpkg and rpm will take care of this for you, no need to use a script here